services/app-api/src/resolvers/healthPlanPackage/healthPlanPackageResolver.ts

```diff
@@ -2,40 +2,63 @@ import { packageStatus, packageSubmittedAt } from '../../domain-models'
 import statePrograms from '../../../../app-web/src/common-code/data/statePrograms.json'
 import { Resolvers } from '../../gen/gqlServer'
+import { isStoreError, Store } from '../../postgres'
+import { convertToIndexQuestionsPayload } from '../../postgres/questionResponse'
+import { logError } from '../../logger'
+import { setErrorAttributesOnActiveSpan } from '../attributeHelper'
 
-export const healthPlanPackageResolver: Resolvers['HealthPlanPackage'] = {
-    revisions(parent) {
-        return parent.revisions.map((r) => {
-            return {
-                node: {
-                    id: r.id,
-                    unlockInfo: r.unlockInfo,
-                    submitInfo: r.submitInfo,
-                    createdAt: r.createdAt,
-                    formDataProto: protoToBase64(r.formDataProto),
-                },
+export function healthPlanPackageResolver(
+    store: Store
+): Resolvers['HealthPlanPackage'] {
+    return {
+        revisions(parent) {
+            return parent.revisions.map((r) => {
+                return {
+                    node: {
+                        id: r.id,
+                        unlockInfo: r.unlockInfo,
+                        submitInfo: r.submitInfo,
+                        createdAt: r.createdAt,
+                        formDataProto: protoToBase64(r.formDataProto),
+                    },
+                }
+            })
+        },
+        status(parent) {
+            const status = packageStatus(parent)
+            if (status instanceof Error) {
+                throw status
             }
-        })
-    },
-    status(parent) {
-        const status = packageStatus(parent)
-        if (status instanceof Error) {
-            throw status
-        }
-        return status
-    },
-    initiallySubmittedAt(parent) {
-        return packageSubmittedAt(parent) || null
-    },
-    state(parent) {
-        const packageState = parent.stateCode
-        const state = statePrograms.states.find(
-            (st) => st.code === packageState
-        )
+            return status
+        },
+        initiallySubmittedAt(parent) {
+            return packageSubmittedAt(parent) || null
+        },
+        state(parent) {
+            const packageState = parent.stateCode
+            const state = statePrograms.states.find(
+                (st) => st.code === packageState
+            )
 
-        if (state === undefined) {
-            throw new Error('State not found in database: ' + packageState)
-        }
-        return state
-    },
+            if (state === undefined) {
+                throw new Error('State not found in database: ' + packageState)
+            }
+            return state
+        },
+        questions: async (parent, args, context) => {
+            const { span } = context
+            const pkgID = parent.id
+            const result = await store.findAllQuestionsByHealthPlanPackage(
```

Comment:
if we are always including questions on the health plan package response (which is my understand of the intent behind this code change)  I don't think we have to do this is in the separate call to store in the resolver. 

We can just prisma `include` the questions field on the db level query.  In other words, I would handle this directly in the initial prisma queries - things like findHealthPlanPackage or the findAll functions. That way the error handling also is handled there in an existing place.

Reply:
> if we are always including questions on the health plan package response
This db query for questions only happens if the graphql query includes questions in the query. So queries like `indexHealthPlanPackages` and `fetchHealthPlanPackage` which doesn't won't call that function in the `healthPlanPackage` resolver.

The cool thing is if you wanted questions in those queries you just need to include the question field in those queries without having to go into those resolvers and manually query the db.

Reply:
> We can just prisma include the questions field on the db level query.

We could do this! I actually went down this road until @macrael showed me the light of adding the question field to the `healthPlanPackageResolver`. Also adding this to `fetchHealthPlanPackage` would also grab questions when on the submission form.

Reply:
ohhh I see thats nice. Gotcha! I understand now the benefits. i like that composability element and didn't realize it wouldn't be called every time. sweet

Reply:
@JasonLin0991 I’d love for you to demo this in our next eng hangout meeting, it’s neat!